### PR TITLE
MDEXP-189 Fix NPE for 008 field

### PR DESCRIPTION
**Pull request by andrii-novytskyi**
## Purpose
https://issues.folio.org/browse/MDEXP-189

## Approach
Added checks for fields that are filled in the field '008'.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java [hunk 1]
```diff
@@ -126,16 +128,29 @@ public String apply(String updatedDate, int currentIndex, Translation translatio
   SET_FIXED_LENGTH_DATA_ELEMENTS() {
     private transient DateTimeFormatter originCreatedDateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
     private transient DateTimeFormatter targetCreatedDateFormatter = DateTimeFormatter.ofPattern("yyMMdd");
+    private String datesOfPublicationPattern = "datesOfPublication";
```
Comment (KVupp) [inline src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java:131]: these are constants, please use the proper naming, with capitals and final for more readability

Comment (andrii-novytskyi) [reply src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java:131]: sure, I will push changes now

File: src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java [hunk 2]
```diff
@@ -126,16 +128,29 @@ public String apply(String updatedDate, int currentIndex, Translation translatio
...
+      } else {
+        createdDateParam = targetCreatedDateFormatter.format(ZonedDateTime.now());
+      }
 
       String publicationDate0Param = "||||";
       String publicationDate1Param = "||||";
-      if (metadata != null && metadata.getData().containsKey("datesOfPublication")) {
-        List<String> publicationDates = (List<String>) metadata.getData().get("datesOfPublication").getData();
+      if (metadata != null && metadata.getData().containsKey(DATES_OF_PUBLICATION)
+        && metadata.getData().get(DATES_OF_PUBLICATION) != null) {
```
Comment (KVupp) [inline src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java:152]: I think something is incorrect here:
The publication field looks like this:
```
"publication" : [ {
      "publisher" : "American Bar Association",
      "place" : "Chicago, Ill.",
      "dateOfPublication" : null,
      "role" : null
    } ],
```

look at "dateOfPublication". There is no "date**s**OfPublication". 
Also, it returns an array, you should be checking if each entry is null or empty

Comment (KVupp) [reply src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java:152]: to be more clear, what if the metadata contains:   metadata.addData("datesOfPublication",  [null, 2016]);. Is this case handled?

Comment (andrii-novytskyi) [reply src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java:152]: agree, not handled., will add more checks

**Pull request merged**